### ISSUE TITLE
Cannot download a JSON file from an APIREST

### ISSUE THREAD

**Issue by rocaseres** [aelassas/Wexflow#237]
Good evening:
I have been trying to download a JSON file from an API REST using the new Wexflow 5.6 version that was recently uploaded, but I could not make it work. 
I think the JSON does not download correctly. Do you have any examples that I could use?
Here is the XML file that shows what I did. Thank you in advance.
[rocaseres.txt](https://github.com/aelassas/Wexflow/files/4619408/rocaseres.txt)
![unnamed](https://user-images.githubusercontent.com/51235147/81763218-1a4d2700-94a5-11ea-8574-99d9d1dc9917.png)

**Comment by aelassas** [comment]
@rocaseres I fixed the issue. Can you try the following things:

1. Stop Wexflow service
2. Backup the existing copy of Wexflow.Tasks.HttpPost.dll in C:\Program Files\Wexflow
3. Copy paste the new [Wexflow.Tasks.HttpPost.dll.zip](https://github.com/aelassas/Wexflow/files/4619471/Wexflow.Tasks.HttpPost.dll.zip) in C:\Program Files\Wexflow
4. Start Wexflow service
5. Kick off your workflow

I added a new optional setting called **type** which defaults to application/json.

**Comment by rocaseres** [comment]
Ok, thank you.
I tried to fix my issue, but I cannot make it work. Here is what I did and the error that comes up.
{"detail":"JSON parse error - Expecting value: line 1 column 1 (char 0)"}
Another little comment in case you didn't notice: I tried to use the new optional setting (type), but it does not appear in the Designer. However, I can use it in the XML section. 
Thank you in advance!
[ejemploJSON.TXT](https://github.com/aelassas/Wexflow/files/4624070/ejemploJSON.TXT)

**Comment by aelassas** [comment]
@rocaseres Now I understand your POST request. I updated HttpPost task. Can you please try the following steps:
1. Stop Wexflow service
2. Backup the existing copy of Wexflow.Tasks.HttpPost.dll in C:\Program Files\Wexflow
3. Copy paste the new [Wexflow.Tasks.HttpPost.dll.zip](https://github.com/aelassas/Wexflow/files/4624170/Wexflow.Tasks.HttpPost.dll.zip) in C:\Program Files\Wexflow
4. Start Wexflow Service
5. Update your workflow as follows:
```xml
<Task id="1" name="HttpPost" description="Executing POST request" enabled="true">
	<Setting name="url" value="https://api.instoreview.cl/api/v2/download-zone/sales/" />
	<Setting name="authorizationScheme" value="ISVToken" />
	<Setting name="authorizationParameter" value="d2dtfabcd2d742d4bedac86cb12e7a7f" />
	<Setting name="type" value="application/json" />
	<Setting name="payload" value="{  &quot;view_type&quot;: &quot;diario&quot;,      &quot;dates&quot;: [&quot;2020-05-01&quot;, &quot;2020-05-02&quot;, &quot;2020-05-03&quot;, &quot;2020-05-04&quot;, &quot;2020-05-05&quot;, &quot;2020-05-06&quot;, &quot;2020-05-07&quot;, &quot;2020-05-08&quot;, &quot;2020-05-09&quot;, &quot;2020-05-10&quot;, &quot;2020-05-11&quot;, &quot;2020-05-12&quot;, &quot;2020-05-13&quot;, &quot;2020-05-14&quot;, &quot;2020-05-15&quot;, &quot;2020-05-16&quot;, &quot;2020-05-17&quot;, &quot;2020-05-18&quot;, &quot;2020-05-19&quot;, &quot;2020-05-20&quot;, &quot;2020-05-21&quot;, &quot;2020-05-22&quot;, &quot;2020-05-23&quot;, &quot;2020-05-24&quot;, &quot;2020-05-25&quot;, &quot;2020-05-26&quot;, &quot;2020-05-27&quot;, &quot;2020-05-28&quot;, &quot;2020-05-29&quot;, &quot;2020-05-30&quot;, &quot;2020-05-31&quot;],      &quot;views&quot;: [&quot;Unidades&quot;, &quot;Ventas B2B&quot;],      &quot;chain_codes&quot;: [],      &quot;hierarchy&quot;: {            &quot;Canal&quot;: [&quot;Farmacias&quot;],            &quot;Cadena&quot;: [],            &quot;Sub Cadena&quot;: [],            &quot;Región&quot;: [],            &quot;Local&quot;: [],            &quot;Descripción Producto&quot;: [],            &quot;EAN&quot;: [],            &quot;SKU&quot;: [],            &quot;Outlet Unique Code&quot;: [],            &quot;Outlet Name&quot;: [],            &quot;Marca&quot;: []            }   }" />
</Task>
```
6. Start your workflow

I added authorizationScheme and authorizationParameter settings.

I also fixed an issue in the designer regarding the payload not showing up in the text box, if you want to the fix, follow these steps:

1. Go to C:\Program Files\Wexflow\Backend\js
2. Backup designer.min.js
3. Copy paste the new  [designer.min.js.zip](https://github.com/aelassas/Wexflow/files/4624198/designer.min.js.zip) in js folder

Regarding the new tasks settings, follow these setps:

1. Go to C:\Wexflow
2. Backup TasksSettings.json
3. Copy paste the new [TasksSettings.json.zip](https://github.com/aelassas/Wexflow/files/4624206/TasksSettings.zip) in C:\Wexflow

and the new settings will show up in the designer.

I will try to update HttpPut, HttpDelete and HttpGet as well.

**Comment by rocaseres** [comment]
Thank you for putting so much effort in my issue. You really helped me. It works now!

**Comment by aelassas** [comment]
@rocaseres I released the [v5.7](https://github.com/aelassas/Wexflow/releases/tag/v5.7) today. It contains the fixes.
